// react-bootstrap-editor/dist/index.esm.js
import {jsx as $13t2B$jsx, jsxs as $13t2B$jsxs, Fragment as $13t2B$Fragment} from "react/jsx-runtime";
import {createRef as $13t2B$createRef, PureComponent as $13t2B$PureComponent} from "react";
import {defineProperty as $13t2B$defineProperty} from "@swc/helpers";
import {BoldTool as $13t2B$BoldTool, ItalicTool as $13t2B$ItalicTool, UnderlineTool as $13t2B$UnderlineTool, StrikeThroughTool as $13t2B$StrikeThroughTool, H1Tool as $13t2B$H1Tool, H2Tool as $13t2B$H2Tool, H3Tool as $13t2B$H3Tool, FontSizeDownTool as $13t2B$FontSizeDownTool, FontSizeUpTool as $13t2B$FontSizeUpTool, SubscriptTool as $13t2B$SubscriptTool, SuperscriptTool as $13t2B$SuperscriptTool, LinkTool as $13t2B$LinkTool, ForeColorTool as $13t2B$ForeColorTool, BackColorTool as $13t2B$BackColorTool, AlignLeftTool as $13t2B$AlignLeftTool, AlignCenterTool as $13t2B$AlignCenterTool, AlignRightTool as $13t2B$AlignRightTool, AlignFullTool as $13t2B$AlignFullTool, OrderedListTool as $13t2B$OrderedListTool, UnorderedListTool as $13t2B$UnorderedListTool, HorizontalRuleTool as $13t2B$HorizontalRuleTool, IFrameTool as $13t2B$IFrameTool, ImageTool as $13t2B$ImageTool, AudioTool as $13t2B$AudioTool, VideoTool as $13t2B$VideoTool, UndoTool as $13t2B$UndoTool, RedoTool as $13t2B$RedoTool, ResetTool as $13t2B$ResetTool, ClearTool as $13t2B$ClearTool} from "edkit";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
var $43d566ad2fe8cba8$exports = {};

$parcel$export($43d566ad2fe8cba8$exports, "renderTool", function () { return $43d566ad2fe8cba8$export$56ccecc1dac11363; });


function $43d566ad2fe8cba8$export$56ccecc1dac11363(editor) {
    const { title: title , active: active , icon: icon , usable: usable  } = this;
    const Class = `btn btn-${(active ? '' : 'outline-') + 'secondary'} me-2 mb-2`;
    return /*#__PURE__*/ $13t2B$jsx("button", {
        type: "button",
        title: title,
        className: Class,
        style: {
            cursor: usable ? 'pointer' : 'not-allowed'
        },
        disabled: !usable,
        onClick: (event)=>{
            event.preventDefault();
            if (editor.current) this.execute(editor.current);
        },
        children: /*#__PURE__*/ $13t2B$jsx("i", {
            className: `bi-${icon}`
        })
    }, icon);
}


var $1a9e4a89b237a23e$exports = {};

$parcel$export($1a9e4a89b237a23e$exports, "TextTools", function () { return $1a9e4a89b237a23e$export$e252ba763eff5cdc; });
$parcel$export($1a9e4a89b237a23e$exports, "ColorTools", function () { return $1a9e4a89b237a23e$export$bdef6e97a10dbad4; });
$parcel$export($1a9e4a89b237a23e$exports, "LayoutTools", function () { return $1a9e4a89b237a23e$export$145c1b37c0810d7; });
$parcel$export($1a9e4a89b237a23e$exports, "MediaTools", function () { return $1a9e4a89b237a23e$export$999636c6e25075fb; });
$parcel$export($1a9e4a89b237a23e$exports, "ControlTools", function () { return $1a9e4a89b237a23e$export$f35db4b82786c3a4; });
$parcel$export($1a9e4a89b237a23e$exports, "OriginalTools", function () { return $1a9e4a89b237a23e$export$9ddb256264948609; });
$parcel$export($1a9e4a89b237a23e$exports, "DefaultTools", function () { return $1a9e4a89b237a23e$export$aa718db1466ce312; });
var $eaf87102ba547294$exports = {};

$parcel$export($eaf87102ba547294$exports, "BoldTool", function () { return $eaf87102ba547294$export$10056a49e3b218eb; });
$parcel$export($eaf87102ba547294$exports, "ItalicTool", function () { return $eaf87102ba547294$export$edf99d0f51bacb1e; });
$parcel$export($eaf87102ba547294$exports, "UnderlineTool", function () { return $eaf87102ba547294$export$4b6d9c17188960d2; });
$parcel$export($eaf87102ba547294$exports, "StrikeThroughTool", function () { return $eaf87102ba547294$export$a35bffaeae8fdda0; });
$parcel$export($eaf87102ba547294$exports, "H1Tool", function () { return $eaf87102ba547294$export$30e8c6b428f8749f; });
$parcel$export($eaf87102ba547294$exports, "H2Tool", function () { return $eaf87102ba547294$export$58443375efa91029; });
$parcel$export($eaf87102ba547294$exports, "H3Tool", function () { return $eaf87102ba547294$export$75d24d6b8b747307; });
$parcel$export($eaf87102ba547294$exports, "FontSizeDownTool", function () { return $eaf87102ba547294$export$a20806c6ef6938f8; });
$parcel$export($eaf87102ba547294$exports, "FontSizeUpTool", function () { return $eaf87102ba547294$export$744da7e24e4a7382; });
$parcel$export($eaf87102ba547294$exports, "SubscriptTool", function () { return $eaf87102ba547294$export$6137518b4e998008; });
$parcel$export($eaf87102ba547294$exports, "SuperscriptTool", function () { return $eaf87102ba547294$export$7217f06c61ebd4fc; });
$parcel$export($eaf87102ba547294$exports, "LinkTool", function () { return $eaf87102ba547294$export$f6c54364b5332e66; });



class $eaf87102ba547294$export$10056a49e3b218eb extends $13t2B$BoldTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'type-bold');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$edf99d0f51bacb1e extends $13t2B$ItalicTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'type-italic');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$4b6d9c17188960d2 extends $13t2B$UnderlineTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'type-underline');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$a35bffaeae8fdda0 extends $13t2B$StrikeThroughTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'type-strikethrough');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$30e8c6b428f8749f extends $13t2B$H1Tool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'type-h1');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$58443375efa91029 extends $13t2B$H2Tool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'type-h2');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$75d24d6b8b747307 extends $13t2B$H3Tool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'type-h3');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$a20806c6ef6938f8 extends $13t2B$FontSizeDownTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'sort-alpha-down');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$744da7e24e4a7382 extends $13t2B$FontSizeUpTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'sort-alpha-up');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$6137518b4e998008 extends $13t2B$SubscriptTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'box-arrow-down-right');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$7217f06c61ebd4fc extends $13t2B$SuperscriptTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'box-arrow-up-right');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $eaf87102ba547294$export$f6c54364b5332e66 extends $13t2B$LinkTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'link');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}


var $531f464768a4571c$exports = {};

$parcel$export($531f464768a4571c$exports, "ColorSelector", function () { return $531f464768a4571c$export$9012acaf449cdfcb; });
$parcel$export($531f464768a4571c$exports, "renderColorTool", function () { return $531f464768a4571c$export$e913fac2ab7b6f29; });
$parcel$export($531f464768a4571c$exports, "ForeColorTool", function () { return $531f464768a4571c$export$cab0e15cfb8489d7; });
$parcel$export($531f464768a4571c$exports, "BackColorTool", function () { return $531f464768a4571c$export$92bb6183e7459be2; });




function $531f464768a4571c$export$9012acaf449cdfcb({ className: className , title: title , type: type , value: value1 , onChange: onChange , icon: icon  }) {
    return /*#__PURE__*/ $13t2B$jsxs("span", {
        className: `d-inline-block align-middle position-relative ${className}`,
        title: title,
        children: [
            /*#__PURE__*/ $13t2B$jsx("input", {
                className: "position-absolute w-100 h-100 rounded-3",
                style: {
                    left: 0,
                    top: 0,
                    zIndex: -1
                },
                type: "color",
                value: value1,
                onChange: ({ target: { value: value  }  })=>{
                    return onChange === null || onChange === void 0 ? void 0 : onChange(value);
                }
            }),
            /*#__PURE__*/ $13t2B$jsx("button", {
                className: "btn",
                style: {
                    color: type === 'color' ? value1 : 'lightgray',
                    backgroundColor: type === 'color' ? 'white' : value1,
                    borderColor: value1
                },
                onClick: (event)=>{
                    event.preventDefault();
                    event.currentTarget.previousElementSibling.click();
                },
                children: /*#__PURE__*/ $13t2B$jsx("i", {
                    className: `bi-${icon}`
                })
            })
        ]
    });
}
function $531f464768a4571c$export$e913fac2ab7b6f29(editor) {
    const { icon: icon , name: name , colorName: colorName  } = this;
    return /*#__PURE__*/ $13t2B$jsx($531f464768a4571c$export$9012acaf449cdfcb, {
        className: "me-2 mb-2",
        title: name,
        icon: icon,
        type: colorName,
        value: this.getColor(),
        onChange: (color)=>editor.current && this.execute(editor.current, color)
    }, icon);
}
class $531f464768a4571c$export$cab0e15cfb8489d7 extends $13t2B$ForeColorTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'file-earmark-font');
        $13t2B$defineProperty(this, "render", $531f464768a4571c$export$e913fac2ab7b6f29);
    }
}
class $531f464768a4571c$export$92bb6183e7459be2 extends $13t2B$BackColorTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'file-earmark-font-fill');
        $13t2B$defineProperty(this, "render", $531f464768a4571c$export$e913fac2ab7b6f29);
    }
}


var $e325b99ac86e8052$exports = {};

$parcel$export($e325b99ac86e8052$exports, "AlignLeftTool", function () { return $e325b99ac86e8052$export$42a444ebfdcfe7b9; });
$parcel$export($e325b99ac86e8052$exports, "AlignCenterTool", function () { return $e325b99ac86e8052$export$9a2e6e0294a984eb; });
$parcel$export($e325b99ac86e8052$exports, "AlignRightTool", function () { return $e325b99ac86e8052$export$22363f91860fc258; });
$parcel$export($e325b99ac86e8052$exports, "AlignFullTool", function () { return $e325b99ac86e8052$export$42827b1362fed748; });
$parcel$export($e325b99ac86e8052$exports, "OrderedListTool", function () { return $e325b99ac86e8052$export$faf4e70f09f43a15; });
$parcel$export($e325b99ac86e8052$exports, "UnorderedListTool", function () { return $e325b99ac86e8052$export$cf4030b57a3fbe37; });
$parcel$export($e325b99ac86e8052$exports, "HorizontalRuleTool", function () { return $e325b99ac86e8052$export$dce8a7d6279fc201; });



class $e325b99ac86e8052$export$42a444ebfdcfe7b9 extends $13t2B$AlignLeftTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'text-left');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $e325b99ac86e8052$export$9a2e6e0294a984eb extends $13t2B$AlignCenterTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'text-center');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $e325b99ac86e8052$export$22363f91860fc258 extends $13t2B$AlignRightTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'text-right');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $e325b99ac86e8052$export$42827b1362fed748 extends $13t2B$AlignFullTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'justify');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $e325b99ac86e8052$export$faf4e70f09f43a15 extends $13t2B$OrderedListTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'list-ol');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $e325b99ac86e8052$export$cf4030b57a3fbe37 extends $13t2B$UnorderedListTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'list-ul');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $e325b99ac86e8052$export$dce8a7d6279fc201 extends $13t2B$HorizontalRuleTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'reception-0');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}


var $fe0ddf31161b8e51$exports = {};

$parcel$export($fe0ddf31161b8e51$exports, "IFrameTool", function () { return $fe0ddf31161b8e51$export$3e557574863db086; });
$parcel$export($fe0ddf31161b8e51$exports, "ImageTool", function () { return $fe0ddf31161b8e51$export$982f47444a2f88da; });
$parcel$export($fe0ddf31161b8e51$exports, "AudioTool", function () { return $fe0ddf31161b8e51$export$6024938f37e646f8; });
$parcel$export($fe0ddf31161b8e51$exports, "VideoTool", function () { return $fe0ddf31161b8e51$export$932111b8028f79ed; });



class $fe0ddf31161b8e51$export$3e557574863db086 extends $13t2B$IFrameTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'window');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $fe0ddf31161b8e51$export$982f47444a2f88da extends $13t2B$ImageTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'image');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $fe0ddf31161b8e51$export$6024938f37e646f8 extends $13t2B$AudioTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'voicemail');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $fe0ddf31161b8e51$export$932111b8028f79ed extends $13t2B$VideoTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'camera-video');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}


var $c165247acc4b6a48$exports = {};

$parcel$export($c165247acc4b6a48$exports, "UndoTool", function () { return $c165247acc4b6a48$export$65235875731ff287; });
$parcel$export($c165247acc4b6a48$exports, "RedoTool", function () { return $c165247acc4b6a48$export$b60f30d254d1e1a6; });
$parcel$export($c165247acc4b6a48$exports, "ResetTool", function () { return $c165247acc4b6a48$export$6f46dc3d81fbb4bd; });
$parcel$export($c165247acc4b6a48$exports, "ClearTool", function () { return $c165247acc4b6a48$export$25eb377ce1feaada; });



class $c165247acc4b6a48$export$65235875731ff287 extends $13t2B$UndoTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'arrow-counterclockwise');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $c165247acc4b6a48$export$b60f30d254d1e1a6 extends $13t2B$RedoTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'arrow-clockwise');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $c165247acc4b6a48$export$6f46dc3d81fbb4bd extends $13t2B$ResetTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'eraser');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}
class $c165247acc4b6a48$export$25eb377ce1feaada extends $13t2B$ClearTool {
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "icon", 'file-earmark-x');
        $13t2B$defineProperty(this, "render", $43d566ad2fe8cba8$export$56ccecc1dac11363);
    }
}







const $1a9e4a89b237a23e$export$e252ba763eff5cdc = [
    $eaf87102ba547294$export$10056a49e3b218eb,
    $eaf87102ba547294$export$edf99d0f51bacb1e,
    $eaf87102ba547294$export$4b6d9c17188960d2,
    $eaf87102ba547294$export$a35bffaeae8fdda0,
    $eaf87102ba547294$export$30e8c6b428f8749f,
    $eaf87102ba547294$export$58443375efa91029,
    $eaf87102ba547294$export$75d24d6b8b747307,
    $eaf87102ba547294$export$a20806c6ef6938f8,
    $eaf87102ba547294$export$744da7e24e4a7382,
    $eaf87102ba547294$export$6137518b4e998008,
    $eaf87102ba547294$export$7217f06c61ebd4fc,
    $eaf87102ba547294$export$f6c54364b5332e66
];
const $1a9e4a89b237a23e$export$bdef6e97a10dbad4 = [
    $531f464768a4571c$export$cab0e15cfb8489d7,
    $531f464768a4571c$export$92bb6183e7459be2
];
const $1a9e4a89b237a23e$export$145c1b37c0810d7 = [
    $e325b99ac86e8052$export$42a444ebfdcfe7b9,
    $e325b99ac86e8052$export$9a2e6e0294a984eb,
    $e325b99ac86e8052$export$22363f91860fc258,
    $e325b99ac86e8052$export$42827b1362fed748,
    $e325b99ac86e8052$export$faf4e70f09f43a15,
    $e325b99ac86e8052$export$cf4030b57a3fbe37,
    $e325b99ac86e8052$export$dce8a7d6279fc201
];
const $1a9e4a89b237a23e$export$999636c6e25075fb = [
    $fe0ddf31161b8e51$export$3e557574863db086,
    $fe0ddf31161b8e51$export$982f47444a2f88da,
    $fe0ddf31161b8e51$export$6024938f37e646f8,
    $fe0ddf31161b8e51$export$932111b8028f79ed
];
const $1a9e4a89b237a23e$export$f35db4b82786c3a4 = [
    $c165247acc4b6a48$export$65235875731ff287,
    $c165247acc4b6a48$export$b60f30d254d1e1a6,
    $c165247acc4b6a48$export$6f46dc3d81fbb4bd,
    $c165247acc4b6a48$export$25eb377ce1feaada
];
const $1a9e4a89b237a23e$export$9ddb256264948609 = [
    ...$1a9e4a89b237a23e$export$e252ba763eff5cdc,
    ...$1a9e4a89b237a23e$export$bdef6e97a10dbad4,
    ...$1a9e4a89b237a23e$export$145c1b37c0810d7,
    ...$1a9e4a89b237a23e$export$999636c6e25075fb,
    ...$1a9e4a89b237a23e$export$f35db4b82786c3a4
];
const $1a9e4a89b237a23e$export$aa718db1466ce312 = [
    $eaf87102ba547294$export$10056a49e3b218eb,
    $eaf87102ba547294$export$edf99d0f51bacb1e,
    $eaf87102ba547294$export$4b6d9c17188960d2,
    $eaf87102ba547294$export$a35bffaeae8fdda0,
    $eaf87102ba547294$export$30e8c6b428f8749f,
    $eaf87102ba547294$export$58443375efa91029,
    $eaf87102ba547294$export$75d24d6b8b747307,
    $eaf87102ba547294$export$6137518b4e998008,
    $eaf87102ba547294$export$7217f06c61ebd4fc,
    $531f464768a4571c$export$cab0e15cfb8489d7,
    $531f464768a4571c$export$92bb6183e7459be2,
    $e325b99ac86e8052$export$42a444ebfdcfe7b9,
    $e325b99ac86e8052$export$9a2e6e0294a984eb,
    $e325b99ac86e8052$export$22363f91860fc258,
    $e325b99ac86e8052$export$42827b1362fed748,
    $e325b99ac86e8052$export$faf4e70f09f43a15,
    $e325b99ac86e8052$export$cf4030b57a3fbe37,
    $e325b99ac86e8052$export$dce8a7d6279fc201,
    $fe0ddf31161b8e51$export$982f47444a2f88da,
    $c165247acc4b6a48$export$65235875731ff287,
    $c165247acc4b6a48$export$b60f30d254d1e1a6,
    $c165247acc4b6a48$export$25eb377ce1feaada
];
$parcel$exportWildcard($1a9e4a89b237a23e$exports, $eaf87102ba547294$exports);
$parcel$exportWildcard($1a9e4a89b237a23e$exports, $531f464768a4571c$exports);
$parcel$exportWildcard($1a9e4a89b237a23e$exports, $e325b99ac86e8052$exports);
$parcel$exportWildcard($1a9e4a89b237a23e$exports, $fe0ddf31161b8e51$exports);
$parcel$exportWildcard($1a9e4a89b237a23e$exports, $c165247acc4b6a48$exports);


var $de23b760ca33f560$exports = {};

$parcel$export($de23b760ca33f560$exports, "Editor", function () { return $de23b760ca33f560$export$7cda8d932e2f33c0; });




class $de23b760ca33f560$export$7cda8d932e2f33c0 extends $13t2B$PureComponent {
    static getDerivedStateFromProps({ tools: tools = $1a9e4a89b237a23e$export$aa718db1466ce312 , value: value  }, { toolList: toolList , data: data  }) {
        return {
            toolList: toolList[0] ? toolList : tools.map((ToolButton)=>new ToolButton()
            ),
            data: data || value
        };
    }
    componentDidMount() {
        document.addEventListener('selectionchange', this.updateTools);
    }
    componentWillUnmount() {
        document.removeEventListener('selectionchange', this.updateTools);
    }
    render() {
        const { toolList: toolList , data: data  } = this.state, { onChange: onChange  } = this.props;
        return /*#__PURE__*/ $13t2B$jsxs($13t2B$Fragment, {
            children: [
                /*#__PURE__*/ $13t2B$jsx("header", {
                    children: toolList.map((tool)=>tool.render(this.box)
                    )
                }),
                /*#__PURE__*/ $13t2B$jsx("div", {
                    ref: this.box,
                    className: "form-control h-auto",
                    contentEditable: true,
                    dangerouslySetInnerHTML: {
                        __html: data
                    },
                    onInput: ({ target: target  })=>{
                        return onChange === null || onChange === void 0 ? void 0 : onChange(target.innerHTML);
                    }
                })
            ]
        });
    }
    constructor(...args){
        super(...args);
        $13t2B$defineProperty(this, "box", /*#__PURE__*/ $13t2B$createRef());
        $13t2B$defineProperty(this, "state", {
            toolList: [],
            data: ''
        });
        $13t2B$defineProperty(this, "updateTools", ()=>{
            if (this.box.current === document.activeElement) this.setState({
                toolList: [
                    ...this.state.toolList
                ]
            });
        });
    }
}
$13t2B$defineProperty($de23b760ca33f560$export$7cda8d932e2f33c0, "displayName", 'Editor');




export {$43d566ad2fe8cba8$export$56ccecc1dac11363 as renderTool, $1a9e4a89b237a23e$export$e252ba763eff5cdc as TextTools, $1a9e4a89b237a23e$export$bdef6e97a10dbad4 as ColorTools, $1a9e4a89b237a23e$export$145c1b37c0810d7 as LayoutTools, $1a9e4a89b237a23e$export$999636c6e25075fb as MediaTools, $1a9e4a89b237a23e$export$f35db4b82786c3a4 as ControlTools, $1a9e4a89b237a23e$export$9ddb256264948609 as OriginalTools, $1a9e4a89b237a23e$export$aa718db1466ce312 as DefaultTools, $eaf87102ba547294$export$10056a49e3b218eb as BoldTool, $eaf87102ba547294$export$edf99d0f51bacb1e as ItalicTool, $eaf87102ba547294$export$4b6d9c17188960d2 as UnderlineTool, $eaf87102ba547294$export$a35bffaeae8fdda0 as StrikeThroughTool, $eaf87102ba547294$export$30e8c6b428f8749f as H1Tool, $eaf87102ba547294$export$58443375efa91029 as H2Tool, $eaf87102ba547294$export$75d24d6b8b747307 as H3Tool, $eaf87102ba547294$export$a20806c6ef6938f8 as FontSizeDownTool, $eaf87102ba547294$export$744da7e24e4a7382 as FontSizeUpTool, $eaf87102ba547294$export$6137518b4e998008 as SubscriptTool, $eaf87102ba547294$export$7217f06c61ebd4fc as SuperscriptTool, $eaf87102ba547294$export$f6c54364b5332e66 as LinkTool, $531f464768a4571c$export$9012acaf449cdfcb as ColorSelector, $531f464768a4571c$export$e913fac2ab7b6f29 as renderColorTool, $531f464768a4571c$export$cab0e15cfb8489d7 as ForeColorTool, $531f464768a4571c$export$92bb6183e7459be2 as BackColorTool, $e325b99ac86e8052$export$42a444ebfdcfe7b9 as AlignLeftTool, $e325b99ac86e8052$export$9a2e6e0294a984eb as AlignCenterTool, $e325b99ac86e8052$export$22363f91860fc258 as AlignRightTool, $e325b99ac86e8052$export$42827b1362fed748 as AlignFullTool, $e325b99ac86e8052$export$faf4e70f09f43a15 as OrderedListTool, $e325b99ac86e8052$export$cf4030b57a3fbe37 as UnorderedListTool, $e325b99ac86e8052$export$dce8a7d6279fc201 as HorizontalRuleTool, $fe0ddf31161b8e51$export$3e557574863db086 as IFrameTool, $fe0ddf31161b8e51$export$982f47444a2f88da as ImageTool, $fe0ddf31161b8e51$export$6024938f37e646f8 as AudioTool, $fe0ddf31161b8e51$export$932111b8028f79ed as VideoTool, $c165247acc4b6a48$export$65235875731ff287 as UndoTool, $c165247acc4b6a48$export$b60f30d254d1e1a6 as RedoTool, $c165247acc4b6a48$export$6f46dc3d81fbb4bd as ResetTool, $c165247acc4b6a48$export$25eb377ce1feaada as ClearTool, $de23b760ca33f560$export$7cda8d932e2f33c0 as Editor};
