// edkit/dist/index.js
var $4Pi8L$swchelpers = require("@swc/helpers");
var $4Pi8L$regeneratorruntime = require("regenerator-runtime");
var $4Pi8L$browserfsaccess = require("browser-fs-access");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $5ddc77f84d52e9cd$exports = {};

$parcel$export($5ddc77f84d52e9cd$exports, "getAnchorElement", function () { return $5ddc77f84d52e9cd$export$ac81f1ba58a5f363; });
$parcel$export($5ddc77f84d52e9cd$exports, "Tool", function () { return $5ddc77f84d52e9cd$export$724f6b867059ef5f; });
$parcel$export($5ddc77f84d52e9cd$exports, "FileTool", function () { return $5ddc77f84d52e9cd$export$de3122b2484864a7; });



function $5ddc77f84d52e9cd$export$ac81f1ba58a5f363() {
    var anchorNode = (self.getSelection() || {}).anchorNode;
    if (anchorNode) return anchorNode.parentElement;
}
var $5ddc77f84d52e9cd$export$724f6b867059ef5f = /*#__PURE__*/ function() {
    "use strict";
    function $5ddc77f84d52e9cd$export$724f6b867059ef5f() {
        $4Pi8L$swchelpers.classCallCheck(this, $5ddc77f84d52e9cd$export$724f6b867059ef5f);
    }
    $4Pi8L$swchelpers.createClass($5ddc77f84d52e9cd$export$724f6b867059ef5f, [
        {
            key: "usable",
            get: function get() {
                var command = this.command;
                return !command || document.queryCommandSupported(command) || document.queryCommandEnabled(command);
            }
        },
        {
            key: "getActive",
            value: function getActive() {
                var ref = this, command = ref.command, tags = ref.tags;
                if (tags) {
                    var box = $5ddc77f84d52e9cd$export$ac81f1ba58a5f363();
                    return !!box && box.matches(tags.map(function(tag) {
                        return "".concat(tag, ", ").concat(tag, " *");
                    }).join(', '));
                }
                if (command) return document.queryCommandState(command);
                return false;
            }
        },
        {
            key: "active",
            get: function get() {
                return this.getActive();
            }
        },
        {
            key: "title",
            get: function get() {
                var ref = this, name = ref.name, keys = ref.keys, usable = ref.usable;
                return "".concat(name).concat(usable ? keys ? "\n(".concat(keys.join(' + '), ")") : '' : '\n(not supported)');
            }
        },
        {
            key: "edit",
            value: function edit(editor) {
                for(var _len = arguments.length, data = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++){
                    data[_key - 1] = arguments[_key];
                }
                var _document;
                editor.focus();
                (_document = document).execCommand.apply(_document, [
                    this.command,
                    null
                ].concat($4Pi8L$swchelpers.toConsumableArray(data)));
            }
        },
        {
            key: "execute",
            value: function execute(editor) {
                for(var _len = arguments.length, data = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++){
                    data[_key - 1] = arguments[_key];
                }
                var inputs = this.inputs, values = [];
                if (inputs) {
                    var data1 = inputs.map(function(input) {
                        return self.prompt(input);
                    });
                    if (!data1.filter(Boolean)[0]) return;
                    values = data1;
                }
                this.edit.apply(this, [
                    editor
                ].concat($4Pi8L$swchelpers.toConsumableArray(values)));
            }
        }
    ]);
    return $5ddc77f84d52e9cd$export$724f6b867059ef5f;
}();
var $5ddc77f84d52e9cd$export$de3122b2484864a7 = /*#__PURE__*/ function($5ddc77f84d52e9cd$export$724f6b867059ef5f) {
    "use strict";
    $4Pi8L$swchelpers.inherits($5ddc77f84d52e9cd$export$de3122b2484864a7, $5ddc77f84d52e9cd$export$724f6b867059ef5f);
    var _super = $4Pi8L$swchelpers.createSuper($5ddc77f84d52e9cd$export$de3122b2484864a7);
    function $5ddc77f84d52e9cd$export$de3122b2484864a7() {
        $4Pi8L$swchelpers.classCallCheck(this, $5ddc77f84d52e9cd$export$de3122b2484864a7);
        return _super.apply(this, arguments);
    }
    $4Pi8L$swchelpers.createClass($5ddc77f84d52e9cd$export$de3122b2484864a7, [
        {
            key: "save",
            value: function save(data) {
                return $4Pi8L$swchelpers.asyncToGenerator(($parcel$interopDefault($4Pi8L$regeneratorruntime)).mark(function _callee() {
                    return ($parcel$interopDefault($4Pi8L$regeneratorruntime)).wrap(function _callee$(_ctx) {
                        while(1)switch(_ctx.prev = _ctx.next){
                            case 0:
                                return _ctx.abrupt("return", URL.createObjectURL(data));
                            case 1:
                            case "end":
                                return _ctx.stop();
                        }
                    }, _callee);
                }))();
            }
        },
        {
            key: "codeOf",
            value: function codeOf(path) {
                return path;
            }
        },
        {
            key: "execute",
            value: function execute(editor) {
                var _this = this;
                return $4Pi8L$swchelpers.asyncToGenerator(($parcel$interopDefault($4Pi8L$regeneratorruntime)).mark(function _callee() {
                    var path;
                    return ($parcel$interopDefault($4Pi8L$regeneratorruntime)).wrap(function _callee$(_ctx) {
                        while(1)switch(_ctx.prev = _ctx.next){
                            case 0:
                                if (!self.confirm('Confirm to upload a file, or cancel to input a path.')) {
                                    _ctx.next = 10;
                                    break;
                                }
                                _ctx.t1 = _this;
                                _ctx.next = 4;
                                return $4Pi8L$browserfsaccess.fileOpen();
                            case 4:
                                _ctx.t2 = _ctx.sent;
                                _ctx.next = 7;
                                return _ctx.t1.save.call(_ctx.t1, _ctx.t2);
                            case 7:
                                _ctx.t0 = _ctx.sent;
                                _ctx.next = 11;
                                break;
                            case 10:
                                _ctx.t0 = self.prompt('Path');
                            case 11:
                                path = _ctx.t0;
                                if (path) _this.edit(editor, _this.codeOf(path));
                            case 13:
                            case "end":
                                return _ctx.stop();
                        }
                    }, _callee);
                }))();
            }
        }
    ]);
    return $5ddc77f84d52e9cd$export$de3122b2484864a7;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);


var $9a2bbe5526331901$exports = {};

$parcel$export($9a2bbe5526331901$exports, "BoldTool", function () { return $9a2bbe5526331901$export$10056a49e3b218eb; });
$parcel$export($9a2bbe5526331901$exports, "ItalicTool", function () { return $9a2bbe5526331901$export$edf99d0f51bacb1e; });
$parcel$export($9a2bbe5526331901$exports, "UnderlineTool", function () { return $9a2bbe5526331901$export$4b6d9c17188960d2; });
$parcel$export($9a2bbe5526331901$exports, "StrikeThroughTool", function () { return $9a2bbe5526331901$export$a35bffaeae8fdda0; });
$parcel$export($9a2bbe5526331901$exports, "H1Tool", function () { return $9a2bbe5526331901$export$30e8c6b428f8749f; });
$parcel$export($9a2bbe5526331901$exports, "H2Tool", function () { return $9a2bbe5526331901$export$58443375efa91029; });
$parcel$export($9a2bbe5526331901$exports, "H3Tool", function () { return $9a2bbe5526331901$export$75d24d6b8b747307; });
$parcel$export($9a2bbe5526331901$exports, "FontSizeDownTool", function () { return $9a2bbe5526331901$export$a20806c6ef6938f8; });
$parcel$export($9a2bbe5526331901$exports, "FontSizeUpTool", function () { return $9a2bbe5526331901$export$744da7e24e4a7382; });
$parcel$export($9a2bbe5526331901$exports, "SubscriptTool", function () { return $9a2bbe5526331901$export$6137518b4e998008; });
$parcel$export($9a2bbe5526331901$exports, "SuperscriptTool", function () { return $9a2bbe5526331901$export$7217f06c61ebd4fc; });
$parcel$export($9a2bbe5526331901$exports, "LinkTool", function () { return $9a2bbe5526331901$export$f6c54364b5332e66; });


var $9a2bbe5526331901$export$10056a49e3b218eb = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$10056a49e3b218eb, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$10056a49e3b218eb);
    function $9a2bbe5526331901$export$10056a49e3b218eb() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$10056a49e3b218eb);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Bold');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'b',
            'strong'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "keys", [
            'Ctrl',
            'B'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'bold');
        return _this;
    }
    return $9a2bbe5526331901$export$10056a49e3b218eb;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$edf99d0f51bacb1e = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$edf99d0f51bacb1e, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$edf99d0f51bacb1e);
    function $9a2bbe5526331901$export$edf99d0f51bacb1e() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$edf99d0f51bacb1e);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Italic');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'i',
            'em'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "keys", [
            'Ctrl',
            'I'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'italic');
        return _this;
    }
    return $9a2bbe5526331901$export$edf99d0f51bacb1e;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$4b6d9c17188960d2 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$4b6d9c17188960d2, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$4b6d9c17188960d2);
    function $9a2bbe5526331901$export$4b6d9c17188960d2() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$4b6d9c17188960d2);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Underline');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'u'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "keys", [
            'Ctrl',
            'U'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'underline');
        return _this;
    }
    return $9a2bbe5526331901$export$4b6d9c17188960d2;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$a35bffaeae8fdda0 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$a35bffaeae8fdda0, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$a35bffaeae8fdda0);
    function $9a2bbe5526331901$export$a35bffaeae8fdda0() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$a35bffaeae8fdda0);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Strike through');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            's',
            'del',
            'strike'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'strikeThrough');
        return _this;
    }
    return $9a2bbe5526331901$export$a35bffaeae8fdda0;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$var$HeadingCommand = document.queryCommandSupported('heading') ? 'heading' : 'formatBlock';
var $9a2bbe5526331901$export$30e8c6b428f8749f = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$30e8c6b428f8749f, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$30e8c6b428f8749f);
    function $9a2bbe5526331901$export$30e8c6b428f8749f() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$30e8c6b428f8749f);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'H1');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'h1'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", $9a2bbe5526331901$var$HeadingCommand);
        return _this;
    }
    $4Pi8L$swchelpers.createClass($9a2bbe5526331901$export$30e8c6b428f8749f, [
        {
            key: "execute",
            value: function execute(editor) {
                this.edit(editor, 'H1');
            }
        }
    ]);
    return $9a2bbe5526331901$export$30e8c6b428f8749f;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$58443375efa91029 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$58443375efa91029, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$58443375efa91029);
    function $9a2bbe5526331901$export$58443375efa91029() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$58443375efa91029);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'H2');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'h2'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", $9a2bbe5526331901$var$HeadingCommand);
        return _this;
    }
    $4Pi8L$swchelpers.createClass($9a2bbe5526331901$export$58443375efa91029, [
        {
            key: "execute",
            value: function execute(editor) {
                this.edit(editor, 'H2');
            }
        }
    ]);
    return $9a2bbe5526331901$export$58443375efa91029;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$75d24d6b8b747307 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$75d24d6b8b747307, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$75d24d6b8b747307);
    function $9a2bbe5526331901$export$75d24d6b8b747307() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$75d24d6b8b747307);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'H3');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'h3'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", $9a2bbe5526331901$var$HeadingCommand);
        return _this;
    }
    $4Pi8L$swchelpers.createClass($9a2bbe5526331901$export$75d24d6b8b747307, [
        {
            key: "execute",
            value: function execute(editor) {
                this.edit(editor, 'H3');
            }
        }
    ]);
    return $9a2bbe5526331901$export$75d24d6b8b747307;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$a20806c6ef6938f8 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$a20806c6ef6938f8, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$a20806c6ef6938f8);
    function $9a2bbe5526331901$export$a20806c6ef6938f8() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$a20806c6ef6938f8);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Font Size down');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'small'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'decreaseFontSize');
        return _this;
    }
    return $9a2bbe5526331901$export$a20806c6ef6938f8;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$744da7e24e4a7382 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$744da7e24e4a7382, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$744da7e24e4a7382);
    function $9a2bbe5526331901$export$744da7e24e4a7382() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$744da7e24e4a7382);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Font Size up');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'big'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'increaseFontSize');
        return _this;
    }
    return $9a2bbe5526331901$export$744da7e24e4a7382;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$6137518b4e998008 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$6137518b4e998008, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$6137518b4e998008);
    function $9a2bbe5526331901$export$6137518b4e998008() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$6137518b4e998008);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Subscript');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'sub'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'subscript');
        return _this;
    }
    return $9a2bbe5526331901$export$6137518b4e998008;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$7217f06c61ebd4fc = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$7217f06c61ebd4fc, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$7217f06c61ebd4fc);
    function $9a2bbe5526331901$export$7217f06c61ebd4fc() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$7217f06c61ebd4fc);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Superscript');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'sup'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'superscript');
        return _this;
    }
    return $9a2bbe5526331901$export$7217f06c61ebd4fc;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $9a2bbe5526331901$export$f6c54364b5332e66 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($9a2bbe5526331901$export$f6c54364b5332e66, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($9a2bbe5526331901$export$f6c54364b5332e66);
    function $9a2bbe5526331901$export$f6c54364b5332e66() {
        $4Pi8L$swchelpers.classCallCheck(this, $9a2bbe5526331901$export$f6c54364b5332e66);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Link');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'a'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "inputs", [
            'Path'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'createLink');
        return _this;
    }
    return $9a2bbe5526331901$export$f6c54364b5332e66;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);


var $ceabaccd1f504406$exports = {};

$parcel$export($ceabaccd1f504406$exports, "ColorTool", function () { return $ceabaccd1f504406$export$ba9bbe5370f1bf2f; });
$parcel$export($ceabaccd1f504406$exports, "ForeColorTool", function () { return $ceabaccd1f504406$export$cab0e15cfb8489d7; });
$parcel$export($ceabaccd1f504406$exports, "BackColorTool", function () { return $ceabaccd1f504406$export$92bb6183e7459be2; });


var $ceabaccd1f504406$export$ba9bbe5370f1bf2f = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($ceabaccd1f504406$export$ba9bbe5370f1bf2f, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($ceabaccd1f504406$export$ba9bbe5370f1bf2f);
    function $ceabaccd1f504406$export$ba9bbe5370f1bf2f() {
        $4Pi8L$swchelpers.classCallCheck(this, $ceabaccd1f504406$export$ba9bbe5370f1bf2f);
        return _super.apply(this, arguments);
    }
    $4Pi8L$swchelpers.createClass($ceabaccd1f504406$export$ba9bbe5370f1bf2f, [
        $4Pi8L$swchelpers.defineProperty({
            key: "getColor"
        }, "value", function getColor() {
            var box = $5ddc77f84d52e9cd$export$ac81f1ba58a5f363(), colorName = this.colorName;
            if (!box) return colorName === 'color' ? '#000000' : '#ffffff';
            var ref = getComputedStyle(box), color = ref[colorName];
            var ref1 = $4Pi8L$swchelpers.slicedToArray(color.match(/\d+/g), 3), red = ref1[0], green = ref1[1], blue = ref1[2];
            return '#' + (+red).toString(16).padStart(2, '0') + (+green).toString(16).padStart(2, '0') + (+blue).toString(16).padStart(2, '0');
        }),
        {
            key: "active",
            get: function get() {
                return !!this.getColor();
            }
        },
        {
            key: "execute",
            value: function execute(editor, color) {
                this.edit(editor, color);
            }
        }
    ]);
    return $ceabaccd1f504406$export$ba9bbe5370f1bf2f;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $ceabaccd1f504406$export$cab0e15cfb8489d7 = /*#__PURE__*/ function($ceabaccd1f504406$export$ba9bbe5370f1bf2f) {
    "use strict";
    $4Pi8L$swchelpers.inherits($ceabaccd1f504406$export$cab0e15cfb8489d7, $ceabaccd1f504406$export$ba9bbe5370f1bf2f);
    var _super = $4Pi8L$swchelpers.createSuper($ceabaccd1f504406$export$cab0e15cfb8489d7);
    function $ceabaccd1f504406$export$cab0e15cfb8489d7() {
        $4Pi8L$swchelpers.classCallCheck(this, $ceabaccd1f504406$export$cab0e15cfb8489d7);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Fore Color');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'foreColor');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "colorName", 'color');
        return _this;
    }
    return $ceabaccd1f504406$export$cab0e15cfb8489d7;
}($ceabaccd1f504406$export$ba9bbe5370f1bf2f);
var $ceabaccd1f504406$export$92bb6183e7459be2 = /*#__PURE__*/ function($ceabaccd1f504406$export$ba9bbe5370f1bf2f) {
    "use strict";
    $4Pi8L$swchelpers.inherits($ceabaccd1f504406$export$92bb6183e7459be2, $ceabaccd1f504406$export$ba9bbe5370f1bf2f);
    var _super = $4Pi8L$swchelpers.createSuper($ceabaccd1f504406$export$92bb6183e7459be2);
    function $ceabaccd1f504406$export$92bb6183e7459be2() {
        $4Pi8L$swchelpers.classCallCheck(this, $ceabaccd1f504406$export$92bb6183e7459be2);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Back Color');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'backColor');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "colorName", 'backgroundColor');
        return _this;
    }
    return $ceabaccd1f504406$export$92bb6183e7459be2;
}($ceabaccd1f504406$export$ba9bbe5370f1bf2f);


var $2bb487cfecefdbac$exports = {};

$parcel$export($2bb487cfecefdbac$exports, "AlignTool", function () { return $2bb487cfecefdbac$export$6c1211091eb3b68f; });
$parcel$export($2bb487cfecefdbac$exports, "AlignLeftTool", function () { return $2bb487cfecefdbac$export$42a444ebfdcfe7b9; });
$parcel$export($2bb487cfecefdbac$exports, "AlignCenterTool", function () { return $2bb487cfecefdbac$export$9a2e6e0294a984eb; });
$parcel$export($2bb487cfecefdbac$exports, "AlignRightTool", function () { return $2bb487cfecefdbac$export$22363f91860fc258; });
$parcel$export($2bb487cfecefdbac$exports, "AlignFullTool", function () { return $2bb487cfecefdbac$export$42827b1362fed748; });
$parcel$export($2bb487cfecefdbac$exports, "OrderedListTool", function () { return $2bb487cfecefdbac$export$faf4e70f09f43a15; });
$parcel$export($2bb487cfecefdbac$exports, "UnorderedListTool", function () { return $2bb487cfecefdbac$export$cf4030b57a3fbe37; });
$parcel$export($2bb487cfecefdbac$exports, "HorizontalRuleTool", function () { return $2bb487cfecefdbac$export$dce8a7d6279fc201; });


var $2bb487cfecefdbac$export$6c1211091eb3b68f = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($2bb487cfecefdbac$export$6c1211091eb3b68f, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($2bb487cfecefdbac$export$6c1211091eb3b68f);
    function $2bb487cfecefdbac$export$6c1211091eb3b68f() {
        $4Pi8L$swchelpers.classCallCheck(this, $2bb487cfecefdbac$export$6c1211091eb3b68f);
        return _super.apply(this, arguments);
    }
    $4Pi8L$swchelpers.createClass($2bb487cfecefdbac$export$6c1211091eb3b68f, [
        {
            key: "active",
            get: function get() {
                if ($4Pi8L$swchelpers.get($4Pi8L$swchelpers.getPrototypeOf($2bb487cfecefdbac$export$6c1211091eb3b68f.prototype), "getActive", this).call(this)) return true;
                var box = $5ddc77f84d52e9cd$export$ac81f1ba58a5f363();
                return !!box && getComputedStyle(box).textAlign === this.align;
            }
        }
    ]);
    return $2bb487cfecefdbac$export$6c1211091eb3b68f;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $2bb487cfecefdbac$export$42a444ebfdcfe7b9 = /*#__PURE__*/ function($2bb487cfecefdbac$export$6c1211091eb3b68f) {
    "use strict";
    $4Pi8L$swchelpers.inherits($2bb487cfecefdbac$export$42a444ebfdcfe7b9, $2bb487cfecefdbac$export$6c1211091eb3b68f);
    var _super = $4Pi8L$swchelpers.createSuper($2bb487cfecefdbac$export$42a444ebfdcfe7b9);
    function $2bb487cfecefdbac$export$42a444ebfdcfe7b9() {
        $4Pi8L$swchelpers.classCallCheck(this, $2bb487cfecefdbac$export$42a444ebfdcfe7b9);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Align Left');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'justifyLeft');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "align", 'left');
        return _this;
    }
    return $2bb487cfecefdbac$export$42a444ebfdcfe7b9;
}($2bb487cfecefdbac$export$6c1211091eb3b68f);
var $2bb487cfecefdbac$export$9a2e6e0294a984eb = /*#__PURE__*/ function($2bb487cfecefdbac$export$6c1211091eb3b68f) {
    "use strict";
    $4Pi8L$swchelpers.inherits($2bb487cfecefdbac$export$9a2e6e0294a984eb, $2bb487cfecefdbac$export$6c1211091eb3b68f);
    var _super = $4Pi8L$swchelpers.createSuper($2bb487cfecefdbac$export$9a2e6e0294a984eb);
    function $2bb487cfecefdbac$export$9a2e6e0294a984eb() {
        $4Pi8L$swchelpers.classCallCheck(this, $2bb487cfecefdbac$export$9a2e6e0294a984eb);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Align Center');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'justifyCenter');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "align", 'center');
        return _this;
    }
    return $2bb487cfecefdbac$export$9a2e6e0294a984eb;
}($2bb487cfecefdbac$export$6c1211091eb3b68f);
var $2bb487cfecefdbac$export$22363f91860fc258 = /*#__PURE__*/ function($2bb487cfecefdbac$export$6c1211091eb3b68f) {
    "use strict";
    $4Pi8L$swchelpers.inherits($2bb487cfecefdbac$export$22363f91860fc258, $2bb487cfecefdbac$export$6c1211091eb3b68f);
    var _super = $4Pi8L$swchelpers.createSuper($2bb487cfecefdbac$export$22363f91860fc258);
    function $2bb487cfecefdbac$export$22363f91860fc258() {
        $4Pi8L$swchelpers.classCallCheck(this, $2bb487cfecefdbac$export$22363f91860fc258);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Align Right');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'justifyRight');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "align", 'right');
        return _this;
    }
    return $2bb487cfecefdbac$export$22363f91860fc258;
}($2bb487cfecefdbac$export$6c1211091eb3b68f);
var $2bb487cfecefdbac$export$42827b1362fed748 = /*#__PURE__*/ function($2bb487cfecefdbac$export$6c1211091eb3b68f) {
    "use strict";
    $4Pi8L$swchelpers.inherits($2bb487cfecefdbac$export$42827b1362fed748, $2bb487cfecefdbac$export$6c1211091eb3b68f);
    var _super = $4Pi8L$swchelpers.createSuper($2bb487cfecefdbac$export$42827b1362fed748);
    function $2bb487cfecefdbac$export$42827b1362fed748() {
        $4Pi8L$swchelpers.classCallCheck(this, $2bb487cfecefdbac$export$42827b1362fed748);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Align Full');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'justifyFull');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "align", 'justify');
        return _this;
    }
    return $2bb487cfecefdbac$export$42827b1362fed748;
}($2bb487cfecefdbac$export$6c1211091eb3b68f);
var $2bb487cfecefdbac$export$faf4e70f09f43a15 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($2bb487cfecefdbac$export$faf4e70f09f43a15, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($2bb487cfecefdbac$export$faf4e70f09f43a15);
    function $2bb487cfecefdbac$export$faf4e70f09f43a15() {
        $4Pi8L$swchelpers.classCallCheck(this, $2bb487cfecefdbac$export$faf4e70f09f43a15);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Ordered list');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'ol'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'insertOrderedList');
        return _this;
    }
    return $2bb487cfecefdbac$export$faf4e70f09f43a15;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $2bb487cfecefdbac$export$cf4030b57a3fbe37 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($2bb487cfecefdbac$export$cf4030b57a3fbe37, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($2bb487cfecefdbac$export$cf4030b57a3fbe37);
    function $2bb487cfecefdbac$export$cf4030b57a3fbe37() {
        $4Pi8L$swchelpers.classCallCheck(this, $2bb487cfecefdbac$export$cf4030b57a3fbe37);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Unordered list');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'ul'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'insertUnorderedList');
        return _this;
    }
    return $2bb487cfecefdbac$export$cf4030b57a3fbe37;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $2bb487cfecefdbac$export$dce8a7d6279fc201 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($2bb487cfecefdbac$export$dce8a7d6279fc201, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($2bb487cfecefdbac$export$dce8a7d6279fc201);
    function $2bb487cfecefdbac$export$dce8a7d6279fc201() {
        $4Pi8L$swchelpers.classCallCheck(this, $2bb487cfecefdbac$export$dce8a7d6279fc201);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Horizontal rule');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'hr'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'insertHorizontalRule');
        return _this;
    }
    return $2bb487cfecefdbac$export$dce8a7d6279fc201;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);


var $ee1dbf60aeaab96b$exports = {};

$parcel$export($ee1dbf60aeaab96b$exports, "IFrameTool", function () { return $ee1dbf60aeaab96b$export$3e557574863db086; });
$parcel$export($ee1dbf60aeaab96b$exports, "ImageTool", function () { return $ee1dbf60aeaab96b$export$982f47444a2f88da; });
$parcel$export($ee1dbf60aeaab96b$exports, "AudioTool", function () { return $ee1dbf60aeaab96b$export$6024938f37e646f8; });
$parcel$export($ee1dbf60aeaab96b$exports, "VideoTool", function () { return $ee1dbf60aeaab96b$export$932111b8028f79ed; });


var $ee1dbf60aeaab96b$export$3e557574863db086 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($ee1dbf60aeaab96b$export$3e557574863db086, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($ee1dbf60aeaab96b$export$3e557574863db086);
    function $ee1dbf60aeaab96b$export$3e557574863db086() {
        $4Pi8L$swchelpers.classCallCheck(this, $ee1dbf60aeaab96b$export$3e557574863db086);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Embed Web page');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'iframe'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'insertHTML');
        return _this;
    }
    $4Pi8L$swchelpers.createClass($ee1dbf60aeaab96b$export$3e557574863db086, [
        {
            key: "execute",
            value: function execute(editor) {
                var path = self.prompt('Path');
                if (path) this.edit(editor, '<iframe style="width: 100%; height: 50vh; border: none" src="'.concat(path, '"></iframe>'));
            }
        }
    ]);
    return $ee1dbf60aeaab96b$export$3e557574863db086;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $ee1dbf60aeaab96b$export$982f47444a2f88da = /*#__PURE__*/ function(FileTool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($ee1dbf60aeaab96b$export$982f47444a2f88da, FileTool);
    var _super = $4Pi8L$swchelpers.createSuper($ee1dbf60aeaab96b$export$982f47444a2f88da);
    function $ee1dbf60aeaab96b$export$982f47444a2f88da() {
        $4Pi8L$swchelpers.classCallCheck(this, $ee1dbf60aeaab96b$export$982f47444a2f88da);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Image');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'img'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'insertHTML');
        return _this;
    }
    $4Pi8L$swchelpers.createClass($ee1dbf60aeaab96b$export$982f47444a2f88da, [
        {
            key: "codeOf",
            value: function codeOf(path) {
                return '<img style="max-width: 100%; height: auto" src="'.concat(path, '"/>');
            }
        }
    ]);
    return $ee1dbf60aeaab96b$export$982f47444a2f88da;
}($5ddc77f84d52e9cd$export$de3122b2484864a7);
var $ee1dbf60aeaab96b$export$6024938f37e646f8 = /*#__PURE__*/ function(FileTool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($ee1dbf60aeaab96b$export$6024938f37e646f8, FileTool);
    var _super = $4Pi8L$swchelpers.createSuper($ee1dbf60aeaab96b$export$6024938f37e646f8);
    function $ee1dbf60aeaab96b$export$6024938f37e646f8() {
        $4Pi8L$swchelpers.classCallCheck(this, $ee1dbf60aeaab96b$export$6024938f37e646f8);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Audio');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'audio'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'insertHTML');
        return _this;
    }
    $4Pi8L$swchelpers.createClass($ee1dbf60aeaab96b$export$6024938f37e646f8, [
        {
            key: "codeOf",
            value: function codeOf(path) {
                return '<audio controls src="'.concat(path, '"></audio>');
            }
        }
    ]);
    return $ee1dbf60aeaab96b$export$6024938f37e646f8;
}($5ddc77f84d52e9cd$export$de3122b2484864a7);
var $ee1dbf60aeaab96b$export$932111b8028f79ed = /*#__PURE__*/ function(FileTool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($ee1dbf60aeaab96b$export$932111b8028f79ed, FileTool);
    var _super = $4Pi8L$swchelpers.createSuper($ee1dbf60aeaab96b$export$932111b8028f79ed);
    function $ee1dbf60aeaab96b$export$932111b8028f79ed() {
        $4Pi8L$swchelpers.classCallCheck(this, $ee1dbf60aeaab96b$export$932111b8028f79ed);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Video');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "tags", [
            'video'
        ]);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'insertHTML');
        return _this;
    }
    $4Pi8L$swchelpers.createClass($ee1dbf60aeaab96b$export$932111b8028f79ed, [
        {
            key: "codeOf",
            value: function codeOf(path) {
                return '<video controls src="'.concat(path, '"></video>');
            }
        }
    ]);
    return $ee1dbf60aeaab96b$export$932111b8028f79ed;
}($5ddc77f84d52e9cd$export$de3122b2484864a7);


var $39dd722b123bef71$exports = {};

$parcel$export($39dd722b123bef71$exports, "UndoTool", function () { return $39dd722b123bef71$export$65235875731ff287; });
$parcel$export($39dd722b123bef71$exports, "RedoTool", function () { return $39dd722b123bef71$export$b60f30d254d1e1a6; });
$parcel$export($39dd722b123bef71$exports, "ResetTool", function () { return $39dd722b123bef71$export$6f46dc3d81fbb4bd; });
$parcel$export($39dd722b123bef71$exports, "ClearTool", function () { return $39dd722b123bef71$export$25eb377ce1feaada; });


var $39dd722b123bef71$export$65235875731ff287 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($39dd722b123bef71$export$65235875731ff287, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($39dd722b123bef71$export$65235875731ff287);
    function $39dd722b123bef71$export$65235875731ff287() {
        $4Pi8L$swchelpers.classCallCheck(this, $39dd722b123bef71$export$65235875731ff287);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Undo');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'undo');
        return _this;
    }
    return $39dd722b123bef71$export$65235875731ff287;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $39dd722b123bef71$export$b60f30d254d1e1a6 = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($39dd722b123bef71$export$b60f30d254d1e1a6, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($39dd722b123bef71$export$b60f30d254d1e1a6);
    function $39dd722b123bef71$export$b60f30d254d1e1a6() {
        $4Pi8L$swchelpers.classCallCheck(this, $39dd722b123bef71$export$b60f30d254d1e1a6);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Redo');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'redo');
        return _this;
    }
    return $39dd722b123bef71$export$b60f30d254d1e1a6;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $39dd722b123bef71$export$6f46dc3d81fbb4bd = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($39dd722b123bef71$export$6f46dc3d81fbb4bd, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($39dd722b123bef71$export$6f46dc3d81fbb4bd);
    function $39dd722b123bef71$export$6f46dc3d81fbb4bd() {
        $4Pi8L$swchelpers.classCallCheck(this, $39dd722b123bef71$export$6f46dc3d81fbb4bd);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Reset');
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "command", 'removeFormat');
        return _this;
    }
    return $39dd722b123bef71$export$6f46dc3d81fbb4bd;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);
var $39dd722b123bef71$export$25eb377ce1feaada = /*#__PURE__*/ function(Tool) {
    "use strict";
    $4Pi8L$swchelpers.inherits($39dd722b123bef71$export$25eb377ce1feaada, Tool);
    var _super = $4Pi8L$swchelpers.createSuper($39dd722b123bef71$export$25eb377ce1feaada);
    function $39dd722b123bef71$export$25eb377ce1feaada() {
        $4Pi8L$swchelpers.classCallCheck(this, $39dd722b123bef71$export$25eb377ce1feaada);
        var _this;
        _this = _super.apply(this, arguments);
        $4Pi8L$swchelpers.defineProperty($4Pi8L$swchelpers.assertThisInitialized(_this), "name", 'Clear');
        return _this;
    }
    $4Pi8L$swchelpers.createClass($39dd722b123bef71$export$25eb377ce1feaada, [
        {
            key: "execute",
            value: function execute(editor) {
                editor.focus();
                document.execCommand('selectAll');
                document.execCommand('delete');
            }
        }
    ]);
    return $39dd722b123bef71$export$25eb377ce1feaada;
}($5ddc77f84d52e9cd$export$724f6b867059ef5f);


$parcel$exportWildcard(module.exports, $5ddc77f84d52e9cd$exports);
$parcel$exportWildcard(module.exports, $9a2bbe5526331901$exports);
$parcel$exportWildcard(module.exports, $ceabaccd1f504406$exports);
$parcel$exportWildcard(module.exports, $2bb487cfecefdbac$exports);
$parcel$exportWildcard(module.exports, $ee1dbf60aeaab96b$exports);
$parcel$exportWildcard(module.exports, $39dd722b123bef71$exports);
