// edkit/dist/index.esm.js
import {fileOpen as $QdH60$fileOpen} from "browser-fs-access";
import {defineProperty as $QdH60$defineProperty} from "@swc/helpers";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $306cc79751608d4c$exports = {};

$parcel$export($306cc79751608d4c$exports, "getAnchorElement", function () { return $306cc79751608d4c$export$ac81f1ba58a5f363; });
$parcel$export($306cc79751608d4c$exports, "Tool", function () { return $306cc79751608d4c$export$724f6b867059ef5f; });
$parcel$export($306cc79751608d4c$exports, "FileTool", function () { return $306cc79751608d4c$export$de3122b2484864a7; });

function $306cc79751608d4c$export$ac81f1ba58a5f363() {
    const { anchorNode: anchorNode  } = self.getSelection() || {};
    if (anchorNode) return anchorNode.parentElement;
}
class $306cc79751608d4c$export$724f6b867059ef5f {
    get usable() {
        const { command: command  } = this;
        return !command || document.queryCommandSupported(command) || document.queryCommandEnabled(command);
    }
    getActive() {
        const { command: command , tags: tags  } = this;
        if (tags) {
            const box = $306cc79751608d4c$export$ac81f1ba58a5f363();
            return !!box && box.matches(tags.map((tag)=>`${tag}, ${tag} *`
            ).join(', '));
        }
        if (command) return document.queryCommandState(command);
        return false;
    }
    get active() {
        return this.getActive();
    }
    get title() {
        const { name: name , keys: keys , usable: usable  } = this;
        return `${name}${usable ? keys ? `\n(${keys.join(' + ')})` : '' : '\n(not supported)'}`;
    }
    edit(editor, ...data) {
        editor.focus();
        document.execCommand(this.command, null, ...data);
    }
    execute(editor, ...data) {
        var { inputs: inputs  } = this, values = [];
        if (inputs) {
            const data = inputs.map((input)=>self.prompt(input)
            );
            if (!data.filter(Boolean)[0]) return;
            values = data;
        }
        this.edit(editor, ...values);
    }
}
class $306cc79751608d4c$export$de3122b2484864a7 extends $306cc79751608d4c$export$724f6b867059ef5f {
    async save(data) {
        return URL.createObjectURL(data);
    }
    codeOf(path) {
        return path;
    }
    async execute(editor) {
        const path = self.confirm('Confirm to upload a file, or cancel to input a path.') ? await this.save(await $QdH60$fileOpen()) : self.prompt('Path');
        if (path) this.edit(editor, this.codeOf(path));
    }
}


var $84e1677d856194d7$exports = {};

$parcel$export($84e1677d856194d7$exports, "BoldTool", function () { return $84e1677d856194d7$export$10056a49e3b218eb; });
$parcel$export($84e1677d856194d7$exports, "ItalicTool", function () { return $84e1677d856194d7$export$edf99d0f51bacb1e; });
$parcel$export($84e1677d856194d7$exports, "UnderlineTool", function () { return $84e1677d856194d7$export$4b6d9c17188960d2; });
$parcel$export($84e1677d856194d7$exports, "StrikeThroughTool", function () { return $84e1677d856194d7$export$a35bffaeae8fdda0; });
$parcel$export($84e1677d856194d7$exports, "H1Tool", function () { return $84e1677d856194d7$export$30e8c6b428f8749f; });
$parcel$export($84e1677d856194d7$exports, "H2Tool", function () { return $84e1677d856194d7$export$58443375efa91029; });
$parcel$export($84e1677d856194d7$exports, "H3Tool", function () { return $84e1677d856194d7$export$75d24d6b8b747307; });
$parcel$export($84e1677d856194d7$exports, "FontSizeDownTool", function () { return $84e1677d856194d7$export$a20806c6ef6938f8; });
$parcel$export($84e1677d856194d7$exports, "FontSizeUpTool", function () { return $84e1677d856194d7$export$744da7e24e4a7382; });
$parcel$export($84e1677d856194d7$exports, "SubscriptTool", function () { return $84e1677d856194d7$export$6137518b4e998008; });
$parcel$export($84e1677d856194d7$exports, "SuperscriptTool", function () { return $84e1677d856194d7$export$7217f06c61ebd4fc; });
$parcel$export($84e1677d856194d7$exports, "LinkTool", function () { return $84e1677d856194d7$export$f6c54364b5332e66; });


class $84e1677d856194d7$export$10056a49e3b218eb extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Bold');
        $QdH60$defineProperty(this, "tags", [
            'b',
            'strong'
        ]);
        $QdH60$defineProperty(this, "keys", [
            'Ctrl',
            'B'
        ]);
        $QdH60$defineProperty(this, "command", 'bold');
    }
}
class $84e1677d856194d7$export$edf99d0f51bacb1e extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Italic');
        $QdH60$defineProperty(this, "tags", [
            'i',
            'em'
        ]);
        $QdH60$defineProperty(this, "keys", [
            'Ctrl',
            'I'
        ]);
        $QdH60$defineProperty(this, "command", 'italic');
    }
}
class $84e1677d856194d7$export$4b6d9c17188960d2 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Underline');
        $QdH60$defineProperty(this, "tags", [
            'u'
        ]);
        $QdH60$defineProperty(this, "keys", [
            'Ctrl',
            'U'
        ]);
        $QdH60$defineProperty(this, "command", 'underline');
    }
}
class $84e1677d856194d7$export$a35bffaeae8fdda0 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Strike through');
        $QdH60$defineProperty(this, "tags", [
            's',
            'del',
            'strike'
        ]);
        $QdH60$defineProperty(this, "command", 'strikeThrough');
    }
}
const $84e1677d856194d7$var$HeadingCommand = document.queryCommandSupported('heading') ? 'heading' : 'formatBlock';
class $84e1677d856194d7$export$30e8c6b428f8749f extends $306cc79751608d4c$export$724f6b867059ef5f {
    execute(editor) {
        this.edit(editor, 'H1');
    }
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'H1');
        $QdH60$defineProperty(this, "tags", [
            'h1'
        ]);
        $QdH60$defineProperty(this, "command", $84e1677d856194d7$var$HeadingCommand);
    }
}
class $84e1677d856194d7$export$58443375efa91029 extends $306cc79751608d4c$export$724f6b867059ef5f {
    execute(editor) {
        this.edit(editor, 'H2');
    }
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'H2');
        $QdH60$defineProperty(this, "tags", [
            'h2'
        ]);
        $QdH60$defineProperty(this, "command", $84e1677d856194d7$var$HeadingCommand);
    }
}
class $84e1677d856194d7$export$75d24d6b8b747307 extends $306cc79751608d4c$export$724f6b867059ef5f {
    execute(editor) {
        this.edit(editor, 'H3');
    }
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'H3');
        $QdH60$defineProperty(this, "tags", [
            'h3'
        ]);
        $QdH60$defineProperty(this, "command", $84e1677d856194d7$var$HeadingCommand);
    }
}
class $84e1677d856194d7$export$a20806c6ef6938f8 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Font Size down');
        $QdH60$defineProperty(this, "tags", [
            'small'
        ]);
        $QdH60$defineProperty(this, "command", 'decreaseFontSize');
    }
}
class $84e1677d856194d7$export$744da7e24e4a7382 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Font Size up');
        $QdH60$defineProperty(this, "tags", [
            'big'
        ]);
        $QdH60$defineProperty(this, "command", 'increaseFontSize');
    }
}
class $84e1677d856194d7$export$6137518b4e998008 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Subscript');
        $QdH60$defineProperty(this, "tags", [
            'sub'
        ]);
        $QdH60$defineProperty(this, "command", 'subscript');
    }
}
class $84e1677d856194d7$export$7217f06c61ebd4fc extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Superscript');
        $QdH60$defineProperty(this, "tags", [
            'sup'
        ]);
        $QdH60$defineProperty(this, "command", 'superscript');
    }
}
class $84e1677d856194d7$export$f6c54364b5332e66 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Link');
        $QdH60$defineProperty(this, "tags", [
            'a'
        ]);
        $QdH60$defineProperty(this, "inputs", [
            'Path'
        ]);
        $QdH60$defineProperty(this, "command", 'createLink');
    }
}


var $8459abd5056b83d4$exports = {};

$parcel$export($8459abd5056b83d4$exports, "ColorTool", function () { return $8459abd5056b83d4$export$ba9bbe5370f1bf2f; });
$parcel$export($8459abd5056b83d4$exports, "ForeColorTool", function () { return $8459abd5056b83d4$export$cab0e15cfb8489d7; });
$parcel$export($8459abd5056b83d4$exports, "BackColorTool", function () { return $8459abd5056b83d4$export$92bb6183e7459be2; });


class $8459abd5056b83d4$export$ba9bbe5370f1bf2f extends $306cc79751608d4c$export$724f6b867059ef5f {
    getColor() {
        const box = $306cc79751608d4c$export$ac81f1ba58a5f363(), { colorName: colorName  } = this;
        if (!box) return colorName === 'color' ? '#000000' : '#ffffff';
        const { [colorName]: color  } = getComputedStyle(box);
        const [red, green, blue] = color.match(/\d+/g);
        return '#' + (+red).toString(16).padStart(2, '0') + (+green).toString(16).padStart(2, '0') + (+blue).toString(16).padStart(2, '0');
    }
    get active() {
        return !!this.getColor();
    }
    execute(editor, color) {
        this.edit(editor, color);
    }
}
class $8459abd5056b83d4$export$cab0e15cfb8489d7 extends $8459abd5056b83d4$export$ba9bbe5370f1bf2f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Fore Color');
        $QdH60$defineProperty(this, "command", 'foreColor');
        $QdH60$defineProperty(this, "colorName", 'color');
    }
}
class $8459abd5056b83d4$export$92bb6183e7459be2 extends $8459abd5056b83d4$export$ba9bbe5370f1bf2f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Back Color');
        $QdH60$defineProperty(this, "command", 'backColor');
        $QdH60$defineProperty(this, "colorName", 'backgroundColor');
    }
}


var $ca6c6c0974c8160e$exports = {};

$parcel$export($ca6c6c0974c8160e$exports, "AlignTool", function () { return $ca6c6c0974c8160e$export$6c1211091eb3b68f; });
$parcel$export($ca6c6c0974c8160e$exports, "AlignLeftTool", function () { return $ca6c6c0974c8160e$export$42a444ebfdcfe7b9; });
$parcel$export($ca6c6c0974c8160e$exports, "AlignCenterTool", function () { return $ca6c6c0974c8160e$export$9a2e6e0294a984eb; });
$parcel$export($ca6c6c0974c8160e$exports, "AlignRightTool", function () { return $ca6c6c0974c8160e$export$22363f91860fc258; });
$parcel$export($ca6c6c0974c8160e$exports, "AlignFullTool", function () { return $ca6c6c0974c8160e$export$42827b1362fed748; });
$parcel$export($ca6c6c0974c8160e$exports, "OrderedListTool", function () { return $ca6c6c0974c8160e$export$faf4e70f09f43a15; });
$parcel$export($ca6c6c0974c8160e$exports, "UnorderedListTool", function () { return $ca6c6c0974c8160e$export$cf4030b57a3fbe37; });
$parcel$export($ca6c6c0974c8160e$exports, "HorizontalRuleTool", function () { return $ca6c6c0974c8160e$export$dce8a7d6279fc201; });


class $ca6c6c0974c8160e$export$6c1211091eb3b68f extends $306cc79751608d4c$export$724f6b867059ef5f {
    get active() {
        if (super.getActive()) return true;
        const box = $306cc79751608d4c$export$ac81f1ba58a5f363();
        return !!box && getComputedStyle(box).textAlign === this.align;
    }
}
class $ca6c6c0974c8160e$export$42a444ebfdcfe7b9 extends $ca6c6c0974c8160e$export$6c1211091eb3b68f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Align Left');
        $QdH60$defineProperty(this, "command", 'justifyLeft');
        $QdH60$defineProperty(this, "align", 'left');
    }
}
class $ca6c6c0974c8160e$export$9a2e6e0294a984eb extends $ca6c6c0974c8160e$export$6c1211091eb3b68f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Align Center');
        $QdH60$defineProperty(this, "command", 'justifyCenter');
        $QdH60$defineProperty(this, "align", 'center');
    }
}
class $ca6c6c0974c8160e$export$22363f91860fc258 extends $ca6c6c0974c8160e$export$6c1211091eb3b68f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Align Right');
        $QdH60$defineProperty(this, "command", 'justifyRight');
        $QdH60$defineProperty(this, "align", 'right');
    }
}
class $ca6c6c0974c8160e$export$42827b1362fed748 extends $ca6c6c0974c8160e$export$6c1211091eb3b68f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Align Full');
        $QdH60$defineProperty(this, "command", 'justifyFull');
        $QdH60$defineProperty(this, "align", 'justify');
    }
}
class $ca6c6c0974c8160e$export$faf4e70f09f43a15 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Ordered list');
        $QdH60$defineProperty(this, "tags", [
            'ol'
        ]);
        $QdH60$defineProperty(this, "command", 'insertOrderedList');
    }
}
class $ca6c6c0974c8160e$export$cf4030b57a3fbe37 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Unordered list');
        $QdH60$defineProperty(this, "tags", [
            'ul'
        ]);
        $QdH60$defineProperty(this, "command", 'insertUnorderedList');
    }
}
class $ca6c6c0974c8160e$export$dce8a7d6279fc201 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Horizontal rule');
        $QdH60$defineProperty(this, "tags", [
            'hr'
        ]);
        $QdH60$defineProperty(this, "command", 'insertHorizontalRule');
    }
}


var $3ae781e262cfa361$exports = {};

$parcel$export($3ae781e262cfa361$exports, "IFrameTool", function () { return $3ae781e262cfa361$export$3e557574863db086; });
$parcel$export($3ae781e262cfa361$exports, "ImageTool", function () { return $3ae781e262cfa361$export$982f47444a2f88da; });
$parcel$export($3ae781e262cfa361$exports, "AudioTool", function () { return $3ae781e262cfa361$export$6024938f37e646f8; });
$parcel$export($3ae781e262cfa361$exports, "VideoTool", function () { return $3ae781e262cfa361$export$932111b8028f79ed; });


class $3ae781e262cfa361$export$3e557574863db086 extends $306cc79751608d4c$export$724f6b867059ef5f {
    execute(editor) {
        const path = self.prompt('Path');
        if (path) this.edit(editor, `<iframe style="width: 100%; height: 50vh; border: none" src="${path}"></iframe>`);
    }
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Embed Web page');
        $QdH60$defineProperty(this, "tags", [
            'iframe'
        ]);
        $QdH60$defineProperty(this, "command", 'insertHTML');
    }
}
class $3ae781e262cfa361$export$982f47444a2f88da extends $306cc79751608d4c$export$de3122b2484864a7 {
    codeOf(path) {
        return `<img style="max-width: 100%; height: auto" src="${path}"/>`;
    }
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Image');
        $QdH60$defineProperty(this, "tags", [
            'img'
        ]);
        $QdH60$defineProperty(this, "command", 'insertHTML');
    }
}
class $3ae781e262cfa361$export$6024938f37e646f8 extends $306cc79751608d4c$export$de3122b2484864a7 {
    codeOf(path) {
        return `<audio controls src="${path}"></audio>`;
    }
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Audio');
        $QdH60$defineProperty(this, "tags", [
            'audio'
        ]);
        $QdH60$defineProperty(this, "command", 'insertHTML');
    }
}
class $3ae781e262cfa361$export$932111b8028f79ed extends $306cc79751608d4c$export$de3122b2484864a7 {
    codeOf(path) {
        return `<video controls src="${path}"></video>`;
    }
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Video');
        $QdH60$defineProperty(this, "tags", [
            'video'
        ]);
        $QdH60$defineProperty(this, "command", 'insertHTML');
    }
}


var $65ad161bf0bd24b9$exports = {};

$parcel$export($65ad161bf0bd24b9$exports, "UndoTool", function () { return $65ad161bf0bd24b9$export$65235875731ff287; });
$parcel$export($65ad161bf0bd24b9$exports, "RedoTool", function () { return $65ad161bf0bd24b9$export$b60f30d254d1e1a6; });
$parcel$export($65ad161bf0bd24b9$exports, "ResetTool", function () { return $65ad161bf0bd24b9$export$6f46dc3d81fbb4bd; });
$parcel$export($65ad161bf0bd24b9$exports, "ClearTool", function () { return $65ad161bf0bd24b9$export$25eb377ce1feaada; });


class $65ad161bf0bd24b9$export$65235875731ff287 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Undo');
        $QdH60$defineProperty(this, "command", 'undo');
    }
}
class $65ad161bf0bd24b9$export$b60f30d254d1e1a6 extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Redo');
        $QdH60$defineProperty(this, "command", 'redo');
    }
}
class $65ad161bf0bd24b9$export$6f46dc3d81fbb4bd extends $306cc79751608d4c$export$724f6b867059ef5f {
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Reset');
        $QdH60$defineProperty(this, "command", 'removeFormat');
    }
}
class $65ad161bf0bd24b9$export$25eb377ce1feaada extends $306cc79751608d4c$export$724f6b867059ef5f {
    execute(editor) {
        editor.focus();
        document.execCommand('selectAll');
        document.execCommand('delete');
    }
    constructor(...args){
        super(...args);
        $QdH60$defineProperty(this, "name", 'Clear');
    }
}




export {$306cc79751608d4c$export$ac81f1ba58a5f363 as getAnchorElement, $306cc79751608d4c$export$724f6b867059ef5f as Tool, $306cc79751608d4c$export$de3122b2484864a7 as FileTool, $84e1677d856194d7$export$10056a49e3b218eb as BoldTool, $84e1677d856194d7$export$edf99d0f51bacb1e as ItalicTool, $84e1677d856194d7$export$4b6d9c17188960d2 as UnderlineTool, $84e1677d856194d7$export$a35bffaeae8fdda0 as StrikeThroughTool, $84e1677d856194d7$export$30e8c6b428f8749f as H1Tool, $84e1677d856194d7$export$58443375efa91029 as H2Tool, $84e1677d856194d7$export$75d24d6b8b747307 as H3Tool, $84e1677d856194d7$export$a20806c6ef6938f8 as FontSizeDownTool, $84e1677d856194d7$export$744da7e24e4a7382 as FontSizeUpTool, $84e1677d856194d7$export$6137518b4e998008 as SubscriptTool, $84e1677d856194d7$export$7217f06c61ebd4fc as SuperscriptTool, $84e1677d856194d7$export$f6c54364b5332e66 as LinkTool, $8459abd5056b83d4$export$ba9bbe5370f1bf2f as ColorTool, $8459abd5056b83d4$export$cab0e15cfb8489d7 as ForeColorTool, $8459abd5056b83d4$export$92bb6183e7459be2 as BackColorTool, $ca6c6c0974c8160e$export$6c1211091eb3b68f as AlignTool, $ca6c6c0974c8160e$export$42a444ebfdcfe7b9 as AlignLeftTool, $ca6c6c0974c8160e$export$9a2e6e0294a984eb as AlignCenterTool, $ca6c6c0974c8160e$export$22363f91860fc258 as AlignRightTool, $ca6c6c0974c8160e$export$42827b1362fed748 as AlignFullTool, $ca6c6c0974c8160e$export$faf4e70f09f43a15 as OrderedListTool, $ca6c6c0974c8160e$export$cf4030b57a3fbe37 as UnorderedListTool, $ca6c6c0974c8160e$export$dce8a7d6279fc201 as HorizontalRuleTool, $3ae781e262cfa361$export$3e557574863db086 as IFrameTool, $3ae781e262cfa361$export$982f47444a2f88da as ImageTool, $3ae781e262cfa361$export$6024938f37e646f8 as AudioTool, $3ae781e262cfa361$export$932111b8028f79ed as VideoTool, $65ad161bf0bd24b9$export$65235875731ff287 as UndoTool, $65ad161bf0bd24b9$export$b60f30d254d1e1a6 as RedoTool, $65ad161bf0bd24b9$export$6f46dc3d81fbb4bd as ResetTool, $65ad161bf0bd24b9$export$25eb377ce1feaada as ClearTool};
